src/Compilers/CSharp/Test/Semantic/Diagnostics/GetDiagnosticsTests.cs

```diff
@@ -271,9 +271,12 @@ private static bool DequeueCompilationEvents(AsyncQueue<CompilationEvent> eventQ
                     {
                         var symbol = symbolDeclaredEvent.Symbol;
                         var added = declaredSymbolNames.Add(symbol.Name);
-                        Assert.True(added, "Unexpected multiple symbol declared events for symbol " + symbol);
-                        var method = symbol as Symbols.MethodSymbol;
-                        Assert.Null(method?.PartialDefinitionPart); // we should never get a partial method's implementation part
+                        if (!added)
+                        {
+                            var isPartialMethod = ((Symbols.SourceMemberMethodSymbol)symbol).PartialDefinitionPart != null ||
```

Comment:
>PartialDefinitionPart [](start = 93, length = 21)

It looks like PartialDefinitionPart and PartialImplementationPart are public APIs on MethodSymbol, let's not add dependency on SourceMemberMethodSymbol type since we can change what it covers in the future. Also, is it safe to cast symbol to a method symbol here? It looks like the old code did that in a safe way.

Reply:
sure.